### PR TITLE
Refactor issue list

### PR DESCRIPTION
**Pull request by wxiaoguang**
1. add backend support for filtering "poster" and "assignee"
    * due to the limits, there is no frontend support at the moment
2. rewrite TS code without jquery, now there are 14 jQuery files left:

```
./features/admin/common.ts:1:import $ from 'jquery';
./features/notification.ts:1:import $ from 'jquery';
./features/org-team.ts:1:import $ from 'jquery';
./features/repo-code.ts:1:import $ from 'jquery';
./features/repo-common.ts:1:import $ from 'jquery';
./features/repo-diff.ts:1:import $ from 'jquery';
./features/repo-issue-content.ts:1:import $ from 'jquery';
./features/repo-issue-sidebar.ts:1:import $ from 'jquery';
./features/repo-issue.ts:1:import $ from 'jquery';
./features/repo-legacy.ts:1:import $ from 'jquery';
./features/repo-new.ts:1:import $ from 'jquery';
./features/repo-projects.ts:1:import $ from 'jquery';
./features/repo-settings.ts:1:import $ from 'jquery';
./features/repo-template.ts:1:import $ from 'jquery';
```

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: web_src/js/features/repo-issue-list.ts [hunk 1]
```diff
@@ -49,17 +49,17 @@ function initRepoIssueListCheckboxes() {
...
     e.preventDefault();
 
-    const url = this.getAttribute('data-url');
-    let action = this.getAttribute('data-action');
-    let elementId = this.getAttribute('data-element-id');
-    let issueIDs = [];
+    const url = el.getAttribute('data-url');
+    let action = el.getAttribute('data-action');
+    let elementId = el.getAttribute('data-element-id');
+    const issueIDList = [];
```
Comment (silverwind) [inline web_src/js/features/repo-issue-list.ts]:
```suggestion
    const issueIDList: string[] = [];
```

File: web_src/js/features/repo-issue-list.ts [hunk 2]
```diff
@@ -49,17 +49,17 @@ function initRepoIssueListCheckboxes() {
     syncIssueSelectionState();
   });
 
-  $('.issue-action').on('click', async function (e) {
+  queryElems(document, '.issue-action', (el) => el.addEventListener('click', async (e: MouseEvent) => {
```
Comment (silverwind) [inline web_src/js/features/repo-issue-list.ts]: Would put callback body on own line.

File: templates/repo/issue/filter_list.tmpl [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{$queryLink := QueryBuild "?" "q" $.Keyword "type" $.ViewType "sort" $.SortType "state" $.State "labels" $.SelectLabels "milestone" $.MilestoneID "project" $.ProjectID "assignee" $.AssigneeID "poster" $.PosterID "archived" (Iif $.ShowArchivedLabels NIL)}}
```
Comment (silverwind) [inline templates/repo/issue/filter_list.tmpl]: Would rename `QueryBuild` to `BuildQuery`, sounds more natural.

Comment (wxiaoguang) [reply templates/repo/issue/filter_list.tmpl]: The name is designed intentionally to match the "Query" family.

image

Comment (silverwind) [reply templates/repo/issue/filter_list.tmpl]: But QueryEscape is the name of a golang stdlib function, while QueryBuild is not.

Comment (wxiaoguang) [reply templates/repo/issue/filter_list.tmpl]: They are the same family. And we use our own "queryEscape" function.

And all names I designed in Gitea are all clear enough, and most legacy rewriting and framework-level designs  are done by me, none of them causes any problem.

Comment (silverwind) [reply templates/repo/issue/filter_list.tmpl]: It's okay I guess.

**Review by silverwind — APPROVED** [review]

**Pull request merged**